lib/apartment/active_record/postgresql_adapter.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# rubocop:disable Style/ClassAndModuleChildren
+
+# NOTE: This patch is meant to remove any schema_prefix appart from the ones for
+# excluded models. The schema_prefix would be resolved by apartment's setting
+# of search path
+module Apartment::PostgreSqlAdapterPatch
+  def default_sequence_name(table, _column)
+    res = super
+    schema_prefix = "#{Apartment::Tenant.current}."
+
+    if res&.starts_with?(schema_prefix)
+      if Apartment.excluded_models.none? { |m| m.constantize.table_name == table }
+        res.delete_prefix!(schema_prefix)
+      else
+        res.sub!(schema_prefix, "#{Apartment::Tenant.default_tenant}.")
```

Comment:
This causes extra work when `default_tenant == current`

Reply:
On second thought, this is not strictly correct, because it should happen even if the sequence name is prefixed with some other scheam. No idea if that can happen though :thinking:

Reply:
I thought of that when i was writing this and then forgot to address it. tks

Reply:
I agree w your second concern. addressed it as well